src/Hamiltonians/FroehlichPolaron.jl

```diff
@@ -0,0 +1,151 @@
+"""
+    FroehlichPolaron(addr::OccupationNumberFS; kwargs...)
+    
+
+The Froehlich polaron Hamiltonian for a 1D lattice with ``M`` momentum modes is given by
+
+```math
+H = \\frac{(p̂_f - p)^2}{m} + ωN̂ - v Σₖ(âₖ^† + â_{-k})
+```
+
+where ``p`` is the total momentum and ``p̂_f = \\frac{M}{l}Σ_k k âₖ^† âₖ`` is the momentum operator for the bosons. ``N̂`` is the number operator for the bosons.
+
+
+# Keyword Arguments
+
+* `v=1.0`: the coupling strength, ``v``
+* `mass=1.0`: the particle mass, ``m``
+* `omega=1.0`: the oscillation frequency of the phonons, ``ω``
+* `l=1.0`: the scale parameter of the momentum lattice, ``l``
+* `p=0.0`: the total momentum, ``p``
+* `momentum_cutoff=nothing`: the maximum total momentum allowed for a basis element
+* `mode_cutoff=10.0`: the maximum population in each momentum mode
+
+"""
+struct FroehlichPolaron{
+    T, # eltype
+    M, # number of modes
+    A<:OccupationNumberFS{M}, # address type
+    MC # momentum cutoff indicating type
+} <: AbstractHamiltonian{T}
+    addr::A
+    v::T
+    mass::T
+    omega::T
+    l::T
+    p::T
+    ks::SVector{M,T} # values for k
+    momentum_cutoff::MC
+    mode_cutoff::T
+end
+
+function FroehlichPolaron(
+    addr::OccupationNumberFS;
+    v=1.0,
+    mass=1.0,
+    omega=1.0,
+    l=1.0,
+    p=0.0,
+    momentum_cutoff=nothing,
+    mode_cutoff=10.0,
+)
+    if _exceed_mode_cutoff(mode_cutoff,addr)
+        throw(ArgumentError("Starting address cannot have occupations that exceed mode_cutoff"))
+    end
+    if iszero(l)
+        throw(ArgumentError("l cannot be zero"))
+    end
+
+    M = num_modes(addr) # this is compile-time information
+    if isnothing(momentum_cutoff)
+        v, p, mass, omega, l, mode_cutoff = promote(float(v), float(p), float(mass), float(omega), float(l), float(mode_cutoff))
+    else
+        v, p, mass, omega, l, momentum_cutoff, mode_cutoff = promote(float(v), float(p), float(mass), float(omega), float(l), float(mode_cutoff), float(momentum_cutoff))    
+    end
+
+    step = typeof(v)(2π/M)
+    if isodd(M)
+        start = -π*(1+1/M) + step
+    else
+        start = -π + step
+    end
+    kr = (M/l)*range(start; step = step, length = M)
+    ks = SVector{M}(kr)
+
+    if !isnothing(momentum_cutoff)
+        momentum = dot(ks,onr(addr))
+        if momentum > momentum_cutoff
+            throw(ArgumentError("Starting address has momentum $momentum which cannot exceed momentum_cutoff $momentum_cutoff"))
+        end
+    end
+
+    return FroehlichPolaron(addr, v, mass, omega, l, p, ks, momentum_cutoff, mode_cutoff)
+end
+
+function Base.show(io::IO, h::FroehlichPolaron)
+    println(io, "FroehlichPolaron($(h.addr); v=$(h.v), mass=$(h.mass), omega=$(h.omega), l=$(h.l), p=$(h.p), ")
+    println("momentum_cutoff=$(h.momentum_cutoff), mode_cutoff=$(h.mode_cutoff))")
```

Comment:
```suggestion
    println(io, "momentum_cutoff=$(h.momentum_cutoff), mode_cutoff=$(h.mode_cutoff))")
```
This was an actual bug.